core-bundle/src/Doctrine/Backup/BackupManager.php

```diff
@@ -190,12 +190,10 @@ private function finishWriting($fileHandle, $deflateContext): void
         if ($deflateContext) {
             fwrite($fileHandle, deflate_add($deflateContext, '', ZLIB_FINISH));
         }
+
         fclose($fileHandle);
     }
 
-    /**
-     * @throws BackupManagerException
-     */
```

Comment:
Why is this removed? My IDE wants that to be present so it knows exceptions are handled properly :)

Reply:
We are not using `@throws` annotations, because they are mostly not maintained correctly and therefore outdated.  You might want to adjust your PHPStorm configuration:

<img width="676" alt="" src="https://user-images.githubusercontent.com/1192057/143053141-1c4e815c-c392-4d95-b17b-2feb9cdb35d6.png">

Reply:
We should maybe discuss this again at some point. Annotating exceptions that are thrown as part of the business logic is IMHO good practice. (This should *only* cover custom exceptions).

Reply:
I disagree. PHP is moving away from phpDoc comments for a reason and so should we.

phpDoc comments are very error-prone, because they are not enforceable, and they tend to be forgotten when things are refactored. We had hundreds of wrong phpDoc comments in the code base and I am sure we still have some.

A modern static PHP analyzer will know which exceptions are thrown even without a `@exception` annotation.

Reply:
I get your point about maintainability but it's IMHO also about API design.

Here is an example:
```php
/**
 * @throws CalculationSkippedException if the calculation was skipped
 */
function calculateSomething(Config $config): int {
    if(!$this->validateConfig($config)) {
        throw new InvalidConfigurationException('bad config');
    }

    // …
    throw new CalculationSkippedException();
}
```

How would you use this API? With the annotation it would be clear that one of the exceptions *must always* be handled because it does not indicate an error:

```php
try { 
    $result = $calculator->calculateSomething($myConfig);
}
catch (CalculationSkippedException) {
    $result = 0;
}
```

Reply:
Guys, you are stopping a PR from being merged. Please discuss this separately because it's not related. We don't have it anywhere, so we don't need it here either.

Reply:
🤷‍♂️

Reply:
PHPStorm does not need a phpDoc to handle your example just fine:

<img width="670" alt="" src="https://user-images.githubusercontent.com/1192057/143191073-df2629e2-b1c5-42c1-b43e-28b71880bec2.png">